In [1]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
!cp -r /content/drive/MyDrive/YOLOV11/Datasets/ /content/

In [3]:
!pip install -U gdown ultralytics

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.0/41.0 kB 3.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 78.2 MB/s eta 0:00:00
  Attempting uninstall: gdown
    Found existing installation: gdown 5.2.2
    Uninstalling gdown-5.2.2:
      Successfully uninstalled gdown-5.2.2


# 1. Import Libraries

In [4]:
from ultralytics import YOLO

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


# 2. Training

In [5]:
yolo_yaml_path = "/content/Datasets/yolo_data/data.yaml"

In [6]:
model_1 = YOLO("yolo11m.pt")

In [7]:
model_2 = YOLO("yolo11n.pt")

In [8]:
results_1 = model_1.train(
    data=yolo_yaml_path,
    epochs=100,
    imgsz=640,
    cache=True,
    patience=20,
    plots=True,
    seed=42,
    name="yolo11m_seed42"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=True, cfg=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/Datasets/yolo_data/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dis=6.0, distill_model=None, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11m.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=yolo11m_seed42, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_ma

In [9]:
results_2 = model_2.train(
    data=yolo_yaml_path,
    epochs=100,
    imgsz=640,
    cache=True,
    patience=20,
    plots=True,
    seed=42,
    name="yolo11n_seed42"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=True, cfg=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/Datasets/yolo_data/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dis=6.0, distill_model=None, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=yolo11n_seed42, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_ma

# 3. Evaluation

In [10]:
model_1_path = "/content/runs/detect/yolo11m_seed42/weights/best.pt"
model_2_path = "/content/runs/detect/yolo11n_seed42/weights/best.pt"

In [13]:
model_1 = YOLO(model_1_path)

metrics_1 = model_1.val(
    data="Datasets/yolo_data/data.yaml",
    split="val"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
YOLO11m summary (fused): 126 layers, 20,030,803 parameters, 0 gradients, 67.6 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2422.9±639.5 MB/s, size: 173.7 KB)
val: Scanning /content/Datasets/yolo_data/val/labels.cache... 43 images, 2 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 43/43 12.0Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 3/3 1.5it/s 1.9s
                   all         43        200       0.87      0.925      0.913      0.713
Speed: 6.5ms preprocess, 25.6ms inference, 0.0ms loss, 1.3ms postprocess per image
Results saved to /content/runs/detect/val-3


In [14]:
metrics_1 = model_1.val(
    data="Datasets/yolo_data/data.yaml",
    split="test"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
YOLO11m summary (fused): 126 layers, 20,030,803 parameters, 0 gradients, 67.6 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2417.0±716.4 MB/s, size: 388.7 KB)
val: Scanning /content/Datasets/yolo_data/test/labels.cache... 7 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 7/7 2.0Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 1.6it/s 0.6s
                   all          7         62      0.945      0.831      0.886      0.598
Speed: 1.8ms preprocess, 34.8ms inference, 0.0ms loss, 2.9ms postprocess per image
Results saved to /content/runs/detect/val-4


In [15]:
model_2 = YOLO(model_2_path)

metrics_2 = model_2.val(
    data="Datasets/yolo_data/data.yaml",
    split="val"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
YOLO11n summary (fused): 101 layers, 2,582,347 parameters, 0 gradients, 6.3 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2230.0±528.2 MB/s, size: 92.0 KB)
val: Scanning /content/Datasets/yolo_data/val/labels.cache... 43 images, 2 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 43/43 15.0Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 3/3 1.8it/s 1.7s
                   all         43        200      0.902      0.874       0.91      0.701
Speed: 6.5ms preprocess, 14.0ms inference, 0.0ms loss, 2.0ms postprocess per image
Results saved to /content/runs/detect/val-5


In [17]:
metrics_2 = model_2.val(
    data="Datasets/yolo_data/data.yaml",
    split="test"
)

Ultralytics 8.4.82 🚀 Python-3.12.13 torch-2.11.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1310.8±237.5 MB/s, size: 386.4 KB)
val: Scanning /content/Datasets/yolo_data/test/labels.cache... 7 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 7/7 1.5Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.3it/s 0.4s
                   all          7         62      0.863      0.506      0.649      0.424
Speed: 1.7ms preprocess, 25.3ms inference, 0.0ms loss, 2.4ms postprocess per image
Results saved to /content/runs/detect/val-7


| Metric    | YOLO11m (Val) | YOLO11n (Val) | YOLO11m (Test) | YOLO11n (Test) |
| --------- | ------------: | ------------: | -------------: | -------------: |
| Precision |         0.870 |     **0.902** |      **0.945** |          0.863 |
| Recall    |     **0.925** |         0.874 |      **0.831** |          0.506 |
| mAP50     |     **0.913** |         0.910 |      **0.886** |          0.649 |
| mAP50-95  |     **0.713** |         0.701 |      **0.598** |          0.424 |
| Inference |       25.6 ms |   **14.0 ms** |        34.8 ms |    **25.3 ms** |
